src/hooks/useCoinListEditOptions.ts

```diff
@@ -0,0 +1,141 @@
+import { difference } from 'lodash';
+import { useCallback, useMemo, useRef } from 'react';
+import { useMMKVObject } from 'react-native-mmkv';
+import { useDispatch } from 'react-redux';
+import { atom, useRecoilState, useSetRecoilState } from 'recoil';
+import useAccountSettings from './useAccountSettings';
+import actions from '@rainbow-me/helpers/editOptionTypes';
+import { setHiddenCoins as reduxSetHiddenCoins } from '@rainbow-me/redux/editOptions';
+
+const selectedItemsAtom = atom<string[]>({
+  default: [],
+  key: 'selectedItemsAtom',
+});
+
+export default function useCoinListEditOptions() {
+  const { accountAddress: address } = useAccountSettings();
+
+  const setSelectedItems = useSetRecoilState(selectedItemsAtom);
+  const [hiddenCoins = []] = useMMKVObject<string[]>('hidden-coins-' + address);
+
+  const [pinnedCoins = []] = useMMKVObject<string[]>('pinned-coins-' + address);
+  const pushSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => {
+        return [...prev.filter(i => i !== item), item];
+      }),
+    [setSelectedItems]
+  );
+
+  const removeSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => [...prev.filter(i => i !== item)]),
+    [setSelectedItems]
+  );
+
+  const toggleSelectedCoin = useCallback(
+    (item: string) =>
+      setSelectedItems(prev => {
+        if (prev.includes(item)) {
+          return [...prev.filter(i => i !== item)];
+        } else {
+          return [...prev, item];
+        }
+      }),
+    [setSelectedItems]
+  );
+
+  const clearSelectedCoins = useCallback(() => setSelectedItems([]), [
+    setSelectedItems,
+  ]);
+
+  return {
+    clearSelectedCoins,
+    hiddenCoins,
+    pinnedCoins,
+    pushSelectedCoin,
+    removeSelectedCoin,
+    toggleSelectedCoin,
+  };
+}
+
+export function useCoinListFinishEditingOptions() {
+  const { accountAddress: address } = useAccountSettings();
```

Comment:
same nit: there seems to be no added benefit of renaming `accountAddress` to `address` and we prefer to use `accountAddress` for clarity when possible